### ISSUE TITLE
GetX 5

### ISSUE THREAD

**Issue by jonataslaw** [jonataslaw/getx#2889]
GetWidget will change in version 5.

I think I have failed to be conservative.

GetX basically hasn't changed since its first version (only the RxTypes have changed due to Flutter updates), and I've been trying to keep breaks to a minimum for each update, after all, this is one of the things I like (or liked) the most in using a framework, not having to tinker with its code for a long time.

However, version 5 became a headache. We were migrating the api to navigator 2 completely without ANY break, this is insane. I had to keep a bunch of code I didn't want to maintain compatibility.

However, the truth is that Flutter has matured over the years, and so have I, and there are many things that I just don't see any sense in anymore.

GetWidget, for example, is a widget that prevented Flutter from updating a widget for performance. This was identical to React's PureComponent, not present in Flutter, however, it complicated the api, and brought more bugs than benefits, simply because it was impossible to modify it. With current applications that have dynamic themes, this has become useless.

Also, it seems that GetX users have come to demonize BuildContext, and well, that might have been my fault, after all, that was GetX's description. However, our role was to provide parallel dependency management approaches, not based on BuildContext, but that didn't mean we stopped using it. Today there are people who use Get.width within widgets, which is a headache, since it does not receive dynamic updates, for that there is context.width, however, despite this being in the documentation, this seems to be ignored by part of the users.

That way, I ended up pushing an update forward, just out of fear, fear of having to refactor my projects, fear of causing too many breakings, and the project went a long time without updates.

However, I think we can be better, I think we can have a solid, modern foundation (the current one was written 4 years ago), and so I think we're going to have some changes.

I created this issue to share all the changes I'm going to make in version 5, to serve as a guide in case something breaks.

This does NOT mean that GetX 5 will be full of breaking changes, just that it will have small breaking changes, like every major package update.

**Comment by jonataslaw** [comment]
1. GetWidget.
GetWidget will NOT prevent further widget updates.
Perhaps it will be merged with GetView, and perhaps it will have one of the following functions:

- Auto track of updates: Not sure yet, but maybe this will be added
- Scopped Widget: It can create a scope to pass a copy of your controller to your children. This controller can be accessed through the BuildContext.

2. Nested Navigation:
Previously we used onGenerateRoute to use nested navigation. This won't work since version 5 uses Navigator 2. We'll use RouterOutlet instead.

3. Whoever uses the "change" immutable api will have some breaks. RxStatus for example will cease to exist.

4. Bindings will change, instead of an empty method you will have to return a List<Binding>. This will allow you to inject the dependency with a scope. Why that? Well, if you have a "ProfileController" controller, and a feed where you can endlessly browse through countless profiles, you'll need something similar. I think the current approach is too "hacking", and you have to use tags to generate dependencies. This usually brings a headache, so let's change.

If you don't use any of the above, the chances that you don't have breaking are high.

Note: We will probably split the package into smaller packages for those who only want to use part of the package. The Get package will continue with all the code, but there will be other smaller packages with only part of the implementation. Each resource will communicate with the others through Middlewares. Ex:

Navigation will trigger events. Dependency injection can listen to these events to decide when to delete a controller.

State management and dependency injection cannot be separated, because the latter depends on the former at any time. However the routing, connection with the api (get_connect), and the animations api, may not be part of the scope of many projects, so let's insert it in GetX, but allow people to use GetX's state manager without they also. Obviously, everything will work better together (since the automatic deletion of controllers integrates well with the routes api), but if someone wants to use it in a different way, that's fine.

**Comment by micodev** [comment]
all the best to you.

**Comment by laeo** [comment]
Version management is very confusing, i can't even find the properly branch or tag or commit id of version 4.6.5.

**Comment by shehabhy1** [comment]
i appreciate your time and effort to provide significant solutions but i have a question for you are you working on this alone ??

**Comment by pye52** [comment]
GetX has already saved a lot of time, and now it's time to use that time to embrace change.

**Comment by ramees-nuhman** [comment]
Thank you, for giving a update and wish you all the best

**Comment by loic-hamdi** [comment]
Thanks for the update and the effort to make this package evolve in the good direction.
Please prepare a migration documentation for the change() api, bindings and navigation.
Thank you @jonataslaw

**Comment by KorayLiman** [comment]
Thanks for trying to keep this great package up to date. We really appreciate your effort.

**Comment by Atlas8981** [comment]
Keep up the great work. Your package help me a lot

**Comment by skylastn** [comment]
you're rock bro

**Comment by the6nnoo** [comment]
go bro....fuck all haters
getx is the best in flutter world!
support getx forever!

**Comment by LoveHeiZhu** [comment]
getx is the best in flutter world!
support getx forever!  
Thank you so much for all you do.

**Comment by ramees-nuhman** [comment]
> I'm finally able to resolve this issue, thank you very much. #2878

Please tell how you managed that.

**Comment by alok2811** [comment]
@jonataslaw  you are the very good and intelligent programmer in the Flutter world
I support getx forever ❤️ 
All the best @jonataslaw

**Comment by Nugrahaz** [comment]
I don't see the update on pub.dev. Is the update not available yet?

**Comment by zhangruiyu** [comment]
separating multiple packages is a really good decision

**Comment by Linkadi98** [comment]
Great to hear that GetX 5 is now on track, our applications are using GetX from start, and it's so easy to new members to catch and understand our codebase because of the simple of this framework, that's why we use GetX over Riverpod, BloC, Provider,...

**Comment by seljad** [comment]
when will the new version be published?

**Comment by gauravmehta13** [comment]
> when will the new version be published?

Next year i guess

**Comment by jonataslaw** [comment]
Hahaha
It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.

I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.

**Comment by Linkadi98** [comment]
> Hahaha
> 
> It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.
> 
> 
> 
> I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.

Noice 🥳

**Comment by alok2811** [comment]
> Hahaha It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.
> 
> I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.

Waiting for new challanges ✌️

**Comment by acquytrenthienduong** [comment]
> Hahaha It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.
> 
> I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.

great

**Comment by han1548772930** [comment]
> 哈哈哈 它几乎准备好了，但我确保它在我的生产项目中完美运行。如果1-3天内没有出现任何问题，它将被推送到pub.dev。
> 
> 我宁愿当小白鼠，也不愿推送更新，你们就是我的，如果出现问题我可以快速修复。

Excited for the update.

**Comment by zhangyutao12** [comment]
> Hahaha It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.
> 
> I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.



> Hahaha It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.
> 
> I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.

go go go

**Comment by seljad** [comment]
> Hahaha It's almost ready to go, but I'm making sure it works perfectly in my projects in production. If it doesn't get any problems within 1-3 days, it will be pushed to pub.dev.
> 
> I'd rather be the guinea pig than push an update and you guys be mine, if something breaks I can fix it quickly.

the best news today

**Comment by qiuyan81** [comment]
day after day~wait for you

**Comment by jefrisingh** [comment]
Looking here again and again / பார்த்த விழி பார்த்தபடி பூத்து இருக்க 👀

**Comment by LoveHeiZhu** [comment]
Looking forward to the push of version 4 fixes

**Comment by zhangruiyu** [comment]
Say 3 days update, but I think longer is acceptable, after all is to release a stable version, unstable release you think it is appropriate? And it's Saturday and I'm sure everyone Posting comments doesn't have to work

**Comment by qiuyan81** [comment]
Can the 4.x version support the latest flutter? There will be bugs when flutter runs after 3.10, thank you for your hard work

**Comment by han1548772930** [comment]
@jonataslaw is there something wrong?

**Comment by cocomanbar** [comment]
Expand a new Findornot function (T? FindOrNot <T> (...)) provided to other pages (that is, the memory is returned, and it will return to empty without the problem of the previous order)

This is a question about global bindding.

Obviously, the global laziness is lazily loaded to register Bindding. It can be easily obtained in any place, which is very convenient.

But some scenarios are my troubles, similar to a kind of misunderstanding. For example, A-PAGE should have binded A-Controller, but because the page logic needs to interact with B-PAGE, B-PAGE will find A get through Find A-Controller.But if it is the first to enter the B-Page, it is obviously unnecessary when dealing with the logic of A-Page, which is obviously unnecessary because the A-Page has not yet been created.This misunderstanding will trigger the additional logic of functions such as onInit OnClose (such as oninit network request, onClose's reset and durable logic, etc.)

**Comment by cocomanbar** [comment]
I encountered some problems during development, which were questions about global binding.
Obviously, the global lazy loading registration binding can be easily obtained on any page, which is very convenient.
But there are some scenarios that bother me, similar to a misoperation. For example, currently A-controller is bound to A-page, and B-controller is bound to B-page. However, because the A-page page logic needs to interact with B-page, B-page A-controller will be found through find to solve the state sharing problem. But if the B-page is entered first, the A-controller will be created by mistake when processing the logic with the A-page. This is obviously unnecessary, because the A-page has not been created yet. This operation may trigger additional onInit onClose and other function execution logic (such as onInit network request, onClose reset persistence logic, etc.)

I hope to expand a new findOrNot function (T? findOrNot<T>(...)) to provide other pages with safe access to Controllers that do not belong to them (similar to return if there is an object created in the memory, and return empty if there is no object). Finally thank you, GetX helped me a lot.

**Comment by zhangruiyu** [comment]
> I encountered some problems during development, which were questions about global binding. Obviously, the global lazy loading registration binding can be easily obtained on any page, which is very convenient. But there are some scenarios that bother me, similar to a misoperation. For example, currently A-controller is bound to A-page, and B-controller is bound to B-page. However, because the A-page page logic needs to interact with B-page, B-page A-controller will be found through find to solve the state sharing problem. But if the B-page is entered first, the A-controller will be created by mistake when processing the logic with the A-page. This is obviously unnecessary, because the A-page has not been created yet. This operation may trigger additional onInit onClose and other function execution logic (such as onInit network request, onClose reset persistence logic, etc.)
> 
> I hope to expand a new findOrNot function (T? findOrNot(...)) to provide other pages with safe access to Controllers that do not belong to them (similar to return if there is an object created in the memory, and return empty if there is no object). Finally thank you, GetX helped me a lot.

```
if(Get.isRegistered<T>()){
      Get.find<T>();
    }
```

**Comment by uzuki-P** [comment]
> I encountered some problems during development, which were questions about global binding. Obviously, the global lazy loading registration binding can be easily obtained on any page, which is very convenient. But there are some scenarios that bother me, similar to a misoperation. For example, currently A-controller is bound to A-page, and B-controller is bound to B-page. However, because the A-page page logic needs to interact with B-page, B-page A-controller will be found through find to solve the state sharing problem. But if the B-page is entered first, the A-controller will be created by mistake when processing the logic with the A-page. This is obviously unnecessary, because the A-page has not been created yet. This operation may trigger additional onInit onClose and other function execution logic (such as onInit network request, onClose reset persistence logic, etc.)
> 
> I hope to expand a new findOrNot function (T? findOrNot(...)) to provide other pages with safe access to Controllers that do not belong to them (similar to return if there is an object created in the memory, and return empty if there is no object). Finally thank you, GetX helped me a lot.

Might be better to use discussion feature next time.

You can use something like this. I put this on my `Utils` file
```dart
/// check if controller is registered or lazyPut is already running
  static bool isControllerRegistered<T extends GetxController>() {
    return Get.isPrepared<T>() || Get.isRegistered<T>();
  }

  /// check if controller is registered or lazyPut is already running
  /// then run callback
  static void isControllerRegisteredThen<T extends GetxController>(
    Function(T ctrl) callback,
  ) {
    if (isControllerRegistered<T>()) {
      callback(Get.find<T>());
    }
  }
```

**Comment by AnonymHK** [comment]
> 它几乎准备好了，但我确保它在我的生产项目中完美运行。如果1-3天内没有出现任何问题，它将被推送到pub.dev。


Looking here again and again.....................................

**Comment by wangjinbiao1985** [comment]
Hi, I have to open this URL more than 5 times a day to follow the latest Getx news and look forward to the new version release.
In fact, I hope the author can fix the bugs of Get4, it's enough for me.
Instantly when Get5 comes out, it still needs to change my project code, and this change will be quite a lot, because now the project is using Get4. Get4 is unstable in flutter V3.10 and later versions.
I can't wait any longer. Because if this goes on, my project won't be deliverable. So we decided to change to "Riverpod".
I am very sorry.
I hope Getx is getting better and better.

**Comment by curlylazy** [comment]
realy thank you for the effort mr jo, GetX realy usefull, and save many my work..

**Comment by FlippedOvO** [comment]
> 

Same on me.
I'll wait 1 more week,
I'll change to RiverPod if Getx still not update

**Comment by Adrift001** [comment]
when will the new version be published?

**Comment by Nugrahaz** [comment]
It's been more than 3 days from the discussed time and strangely I'm still waiting... Haha

**Comment by ruanmax** [comment]
Waiting by the next update... I'm spirit to new version!

**Comment by wph144** [comment]
I'll hold my breath until it's updated

**Comment by jonataslaw** [comment]
Documenting the changes is taking longer than I thought. I had no problems in production. So I'm thinking of releasing it as a release candidate temporarily, until I finish the migration guide.

**Comment by gauravmehta13** [comment]
@jonataslaw  [this](https://github.com/jonataslaw/getx/issues/2903) bug is present in current master, can this be fixed as well with the upcoming RC.

**Comment by jonataslaw** [comment]
Did you try BackButtonListener? I think it should be used. WillPopScope is for the old Navigator 1, maybe it will be deprecated in the future.

**Comment by ruanmax** [comment]
@jonataslaw why you release as stable to show the community what the project is live?

**Comment by han1548772930** [comment]
Will the new version have  error with #2901 ?

**Comment by jonataslaw** [comment]
Hi @han1548772930 I think your cache is corrupted. I tried to reproduce your error, but I could not. 
Could you please try the code below in terminal?

`flutter pub cache repair`

**Comment by jonataslaw** [comment]
> @jonataslaw why you release as stable to show the community what the project is alive?

Because it is a big update, with no migration guide, would be hard to find what change affect your project, and how fix it.

**Comment by zhangruiyu** [comment]
> > @jonataslaw why you release as stable to show the community what the project is alive?
> 
> Because it is a big update, with no migration guide, would be hard to find what change affect your project, and how fix it.

How soon will the next candidate version be released? I can't wait to try it out

**Comment by seljad** [comment]
@jonataslaw will the next update fix this error? [#2888](https://github.com/jonataslaw/getx/issues/2888)

**Comment by LoveHeiZhu** [comment]
> 

Is it possible to update the fix for version 4.6.5 at pub.dev first? Currently there is a delayed logout issue with control.onClose() that causes the view to remain with the previous display data after re-toNamed().
For those who are currently using version 4.6.5, it may be better to continue to use 4.6.5 in order to avoid making too many changes to the project after upgrading to GetX 5.

**Comment by AnonymHK** [comment]
@jonataslaw How long will the new version take? It has far exceeded your previous estimated time!

**Comment by jonataslaw** [comment]
If you want try it, you can add:

```
dependencies:
  get: ^5.0.0-release-candidate-5
```

I am writing the migration guide.

**Comment by alok2811** [comment]
Thank you @jonataslaw  for this Candidate version

**Comment by HandsonOmbiro** [comment]
@jonataslaw, are you working alone in all these?

**Comment by gauravmehta13** [comment]
Thanks for the RC @jonataslaw 
There is one issue that i am facing regarding navigation using Get.to in the release, Added the issue [here](https://github.com/jonataslaw/getx/issues/2911)

**Comment by GotJimmy** [comment]
When I switched to Flutter I checked all the state management solutions out there and concluded that GetX is by far the best -- 3 years later I'm more convinced than ever!! GetX is always the first package I add to a new project -- simplicity & speed wins hands down! Keep up the great work & thank you so much @jonataslaw!

**Comment by GotJimmy** [comment]
@jonataslaw One suggestion for the new animations in v5: a `curve` parameter (for like `ElasticOutCurve()`) would be needed (next to `duration`) to make the animations look better. Straight-line animations just don't cut it always.
Otherwise: adding animations to v5 was another great idea!

**Comment by jonataslaw** [comment]
@GotJimmy thank you for your kinds words e the suggestions. I will include something in this directions in in animations.

**Comment by jonataslaw** [comment]
**Breaking note:**

As GetX 5 will be based entirely on navigator 2, and many people use Get.back() to close dialogs, and other overlays, I am adding a Get.backLegacy() with the same behavior as the old Get.back().
This will allow you to replace Get.back() in bulk with Get.backLegacy() to maintain compatibility with current projects, but remember to migrate to the new api, in GetX 6 Get.back() will be as its name says , just go back to the previous route. 

Get.close() will be responsible for closing overlays, and Get.closeSnackbar() Get.closeAllSnackbars(), Get.closeDialog() and Get.closeAllDialogs() etc, will be used to close the current one, or all overlays of a type only. This will allow, in addition to cleaner and more descriptive code (someone who reads a Get.back() does not always know what it is closing), the possibility of closing one or more overlays of a given type, or just close All overlays without close the currentRoute.

Get.close() in GetX 4 closed routes a certain number of times. This api changed completely in version 5. Get.close() will close overlays, and Get.back() will receive a "times" parameter in which you can pass how many times you want to back (old close behaviour).

This will cause breaking changes in some projects, which is why Get.backLegacy is there, however, I highly advise migrating when you can, and when there is time available for the new api.

**Comment by AnonymHK** [comment]
When will the new version be released?

**Comment by Nugrahaz** [comment]
I didn't expect Getx 6 to come out, I haven't even had a chance to taste Getx 5 haha.

**Comment by amitkumartiwaridis** [comment]
> **Breaking note:**
> 
> As GetX 5 will be based entirely on navigator 2, and many people use Get.back() to close dialogs, and other overlays, I am adding a Get.backLegacy() with the same behavior as the old Get.back(). This will allow you to replace Get.back() in bulk with Get.backLegacy() to maintain compatibility with current projects, but remember to migrate to the new api, in GetX 6 Get.back() will be as its name says , just go back to the previous route.
> 
> Get.close() will be responsible for closing overlays, and Get.closeSnackbar() Get.closeAllSnackbars(), Get.closeDialog() and Get.closeAllDialogs() etc, will be used to close the current one, or all overlays of a type only. This will allow, in addition to cleaner and more descriptive code (someone who reads a Get.back() does not always know what it is closing), the possibility of closing one or more overlays of a given type, or just close All overlays without close the currentRoute.
> 
> Get.close() in GetX 4 closed routes a certain number of times. This api changed completely in version 5. Get.close() will close overlays, and Get.back() will receive a "times" parameter in which you can pass how many times you want to back (old close behaviour).
> 
> This will cause breaking changes in some projects, which is why Get.backLegacy is there, however, I highly advise migrating when you can, and when there is time available for the new api.

Ready to migrate with breaking changes. Waiting for the stable release.

**Comment by colbymaloy** [comment]
most of my issues come when there is no comments. 

For example in GetPage there is 'title' field. What does this do? why is there no comment in the code to describe how to use this. these are my frustrations with getx. Also, odd behavior with navigation sometimes.

**Comment by jonataslaw** [comment]
I will try to complete comments on all public fields. By the way, 'title' is a property present in CupertinoPageRoute, for most people, this parameter is useless (and I was wrong to assume that those who use GetX already have a background in similar resources without a package).
This weekend I'm going to work more on the package.

**Comment by jahnli** [comment]
I like getx, it makes a lot of work easier

**Comment by laterdayi** [comment]
@jonataslaw  When is it expected to be released? Looking forward to it very much

**Comment by laterdayi** [comment]
It's exciting to check every day to see when get5 is coming

**Comment by Linkadi98** [comment]
> It's exciting to check every day to see when get5 is coming
> 
> 

Hi, we all know that Jonata is working hard to release this version and we just need his comments about the release, please don't spam comment like this, we subscribe to this issue and not all of us want to receive email which is not from him every day. Just wait for him to complete his job. Thanks alot!

**Comment by AnonymHK** [comment]
When will the official version be released? I am trying out the beta version, but there are no documentation yet. I used to use Binding like this, but the beta version shows that Binding is outdated. I would like to know how to handle Binding in the beta version? 
 
 class HomeBinding extends Bindings {
  @override
  void dependencies() {
    Get.lazyPut(() => HomeLogic());
  }
}

**Comment by ruanmax** [comment]
@AnonymHK in the version beta updated the name of interface Bimding. 

Use the BindingsInterface:

class HomeBinding extends BindingsInterface { @override void dependencies() { Get.lazyPut(() => HomeLogic()); } }

**Comment by jonataslaw** [comment]
In version 5, Bindings can be scoped, or not scoped (you can have one or several controllers, which are children of a binding, but you will choose the one most appropriate to you due to the context).

I believe this won't change anything in current projects (besides the syntax of the bindings), but it will allow many features in future projects, such as: Having a controller that is responsible for a feed on a user's page, and each user having their own controller scopped, this will avoid problems such as: you navigating to 3 profiles in sequence or displaying the same content (if you use the same controller) or you need tags for this (it ends up making the code more complex).
The context will not be necessary, but it will be available to anyone who wants to use it.

This will also eliminate any and all uses of creating InheritedWidget in the project (for anything) or using Provider, ScoppedModels and the like, since we will have this, only improved (everything mentioned above is vertical, you need to insert it at the top of your tree to have access to the same dependency on two pages, getx is horizontal, you will have the same scope as the provider, but with autodispose when the second page no longer uses the widget).
In other words, if the user is logging in, and uses the LoginRepository, this LoginRepository needs to be allocated forever, even when the user is logged in, using vertical approaches (InheritedWidget, provider, and the like) it will be there in memory, because it is a Widget, and it was allocated at the root of your project.
GetX 5 only does a get for the scoped dependency, and when it ceases to exist, it can no longer be accessed. And memory management is outside the widget tree.

This will not change current projects at all, it will only allow something that closes the gap between: Using context for everything, and not using context for anything, now we have to use the context if I want, and if necessary (such as for resources that require scope) but without giving up the ability to access your controller from anywhere without context, and especially giving up intelligent memory allocation.

**Comment by qiuyan81** [comment]
From expectation to disappointment, I have been waiting for more than a month

**Comment by AClon314** [comment]
Hi, the Beginner Example [Counter App with GetX](https://github.com/jonataslaw/getx#counter-app-with-getx) is not working in 5.0.0-release-candidate-5, maybe this need to update.
>The argument type 'Other' can't be assigned to the parameter type 'Widget Function()'. [argument_type_not_assignable](https://dart.dev/diagnostics/argument_type_not_assignable)

**Comment by laterdayi** [comment]
@jonataslaw  Is there an expected release date?

**Comment by anidotnet** [comment]
@jonataslaw thanks for this wonderful framework. Since I started using it, it has increased my productivity manifold. It helps me to write clear and concise code while separating the concerns.

I have one feature request for GetX 5. The new built-in action and shortcut framework in flutter is very verbose and not user friendly. Applying it to a large desktop application where user can customize key-mapping, seems to be a headache. It is not scalable.

So I was wondering, would it be possible to do something about it in GetX 5?

**Comment by laterdayi** [comment]
@jonataslaw   Looking forward to 5.0, are there any release plans now?

**Comment by AnonymHK** [comment]
I don't want to urge you, but it has been over a month since the original plan. For those who are looking forward to you, it is undoubtedly very disappointing. At least you should explain the reason, rather than letting everyone who supports you wait indefinitely.

**Comment by gauravmehta13** [comment]
@jonataslaw  Given your busy schedule, I'd like to propose a slight shift in focus. Instead of constantly adding new features and potentially introducing instability, I think it would be great if you prioritize keeping the project stable for use.

As you know, thousands of projects depend on GetX, and its stability is crucial for the community. If you're finding it challenging to allocate sufficient time, perhaps you could consider creating a GitHub discussion for maintaining the package. This way, the community can contribute to its upkeep, and it won't solely rely on your availability.

There is already a [GitHub issue](https://github.com/jonataslaw/getx/issues/2679) filed regarding opening the project to new contributors. It seems like a lot of developers are interested in helping out. This could be a fantastic opportunity to tap into the collective resources of the community.

Your expertise and leadership in the project are invaluable, and I believe this shift in strategy will help preserve and enhance the package's reputation within the developer community.

**Comment by yang-lile** [comment]
Getx 5 or 6 are unnecessary. The fundamental problem with getx is poor documentation and the confusion of presentation layer code and business code without context. In fact, this is one problem. If there is a document describing the best practices for using getx, I believe the problem will be easily solved.

**Comment by Apollo108** [comment]
Is it feasible to just fork and maintain it properly, keeping it open to the community?

**Comment by qiuyan81** [comment]
The author has done everything himself. Should he find someone with better skills to help maintain the project, such as some big companies that also use getx? I believe this will make the code healthier.

**Comment by laterdayi** [comment]
Is there no follow-up to this plan  ？

**Comment by nugrahaz** [comment]
Still waiting.... (" 🌀_🌀)

**Comment by DevTello** [comment]
I hope he is still alive and everything is fine with him. Everything could've happened, you know... Is there another way to ping him?

**Comment by fxxzx** [comment]
Didn't you guys get the news? He had a car accident and couldn't make it. RIP.

**Comment by saviourdog** [comment]
> Didn't you guys get the news? He had a car accident and couldn't make it. RIP.

no way!Is he okay?

**Comment by jefrisingh** [comment]
> Didn't you guys get the news? He had a car accident and couldn't make it. RIP.

No, Yesterday he commented on another issue, and stop this non sense

**Comment by fxxzx** [comment]
> > Didn't you guys get the news? He had a car accident and couldn't make it. RIP.
> 
> No, Yesterday he commented on another issue, and stop this non sense

Yes, today he's not with us anymore. So sorry.

**Comment by laterdayi** [comment]
Where did you see that? I hope you're just saying it as a joke

@fxxzx

**Comment by Linkadi98** [comment]
> > > Didn't you guys get the news? He had a car accident and couldn't make it. RIP.
> 
> > 
> 
> > No, Yesterday he commented on another issue, and stop this non sense
> 
> 
> 
> Yes, today he's not with us anymore. So sorry.

Where is your source? If you can't proof it by any links or evidences then stop talking like he is gone now.

**Comment by amitkumartiwaridis** [comment]
> > > Didn't you guys get the news? He had a car accident and couldn't make it. RIP.
> > 
> > 
> > No, Yesterday he commented on another issue, and stop this non sense
> 
> Yes, today he's not with us anymore. So sorry.

6 hours ago he commented on one issue.

**Comment by Mixogen** [comment]
guys pls.
fxxzx - this guy register 11 hour ago.
why u so naive?

**Comment by AnonymHK** [comment]
The author seems to have given up on the release plan for getx5. Can anyone provide a replacement that can be used normally? Thank you very much!

**Comment by GotJimmy** [comment]
> The author seems to have given up on the release plan for getx5. Can anyone provide a replacement that can be used normally? Thank you very much!

Getx 4.6.6 works perfectly in the latest version of Flutter.

**Comment by laterdayi** [comment]
Is there a planned release date?
  @jonataslaw

**Comment by fxxzx** [comment]
> Is there a planned release date? @jonataslaw

No, can't push code from heaven

**Comment by fxxzx** [comment]
RIP getx & @jonataslaw

**Comment by Liangwenb** [comment]
Flutter 3.16.0 has been released. Brother, come on, wait for you to update

**Comment by alok2811** [comment]
@jonataslaw brother pls update what is the status of update 5.0.0

**Comment by jonataslaw** [comment]
Sorry for the delay, I was having some problems ranging from personal issues, to the complete lack of power for 1 week in São Paulo due to rain, which delayed my work, and made it impossible for me to give feedback here.

The candidate release version will be promoted to GetX 5, if anyone has any objections, or any bug in the new version, please let me know within 15 hours, as tomorrow GetX 5 will be released.

I'm building documentation of what I remember that changed, if you could help me, updating your project to the latest release candidate, and pointing out breaks, I would appreciate it, for the way I work, it's ok, but many people use GetX in different ways, ranging from cleanarch, to custom architectures (or none at all haha).

Here is a draft of the breaking changes:

https://github.com/jonataslaw/getx/issues/2966

**Comment by loic-hamdi** [comment]
> Sorry for the delay, I was having some problems ranging from personal issues, to the complete lack of power for 1 week in São Paulo due to rain, which delayed my work, and made it impossible for me to give feedback here.
> 
> The candidate release version will be promoted to GetX 5, if anyone has any objections, or any bug in the new version, please let me know within 15 hours, as tomorrow GetX 5 will be released.
> 
> I'm building documentation of what I remember that changed, if you could help me, updating your project to the latest release candidate, and pointing out breaks, I would appreciate it, for the way I work, it's ok, but many people use GetX in different ways, ranging from cleanarch, to custom architectures (or none at all haha).
> 
> Here is a draft of the breaking changes:
> 
> #2966

Thanks @jonataslaw , keep up the good work bro!

**Comment by anidotnet** [comment]
@jonataslaw thanks for the update. And meantime, can you please block/report @fxxzx for spamming this thread and spreading misinformation?

**Comment by fxxzx** [comment]
wait, what?? it's freedom of speech. your not-wise response on information you receive on internet is not my responsibility.

**Comment by laterdayi** [comment]
> wait, what?? it's freedom of speech. your not-wise response on information you receive on internet is not my responsibility.

fuck you mother

**Comment by laterdayi** [comment]
> wait, what?? it's freedom of speech. your not-wise response on information you receive on internet is not my responsibility.

you mother today must die

**Comment by Linkadi98** [comment]
Guys, don't pollute this thread just because of a random guy and his nonsense comments, this thread is about GetX and it only should talk about GetX. Just ignore him please.


> > wait, what?? it's freedom of speech. your not-wise response on information you receive on internet is not my responsibility.
> 
> you mother today must die

**Comment by DevTello** [comment]
@jonataslaw Hi, any updates on 'when'?

**Comment by fxxzx** [comment]
> @jonataslaw Hi, any updates on 'when'?

no, pls continue dreaming

**Comment by DevTello** [comment]
> When is Getx 6 coming, sir?

Judging from the author's activity - probably never)

**Comment by MahdiForughi** [comment]
i am disappointed.

**Comment by moesaid** [comment]
@jonataslaw looking forward to it, my man! Thanks for all the effort and the time you put in GetX!

**Comment by kauemurakami** [comment]
Em Widgets o context pode  ser usado sem estranhamento.
Creio que a navegação 2.0 foi o maior problema nesse caso, pela vontade de escapar do contexto.

Falando sobre roteamento, eles aproveitaram o navigator ( o mesmo que usavamos para criar uma navegação num bottomnavigator por exemplo), e organizaram com uma hierarquia completamente palpavel a partir de um delegate com pais e filhos, e adicionaram SheelRoute que serve como o nosso navigator citado acima, isso cria a estrutura de path/url (page/detailpage/id=1), e usam o contexto atual para navegar de uma rota a outra.

Creio que isso pode ser facilitado com a estrutura atual do get, pois permitimos childrens no GetRoute, para menos experientes era uma magica sair dando Get.to(route) e poder voltar, mas  a estrutura de path não eram concatenadas se nao configuradas direito, o que faziam muitas pessoa nao implementaram um app com um web, ou dificulta o uso de deeplinks por exemplo.

Mas @jonataslaw você pode olhar pro contexto inicial, criando uma especie de delegate, onde todo app parte da rota **/**, e poderia usar o context global, do proprio MaterialApp.router, pois se navega inicialmente e automaticamente existe algum context, assim você pode gerir todo roteamente a partir do primeiro context com um delegate, e concatenar com um push ou remover a rota atual com o pop  ou caso haja um pushreplacent.

 - exemplo:
Get.push(/rota1)
Get.push(/rotadetails) aqui entao teriamos uma localização atual de /rota1/rotadetails, isso tudo levando em conta o método push
Nesse caso o nosso método pop voltaria de uma em uma, corrigindo a url.

Com um Get.pushreplacement(/rota2) sabemos podemos desempilhar as rotas que estamos e pular para rota2.
Tudo isso com o contexto inicial, levando em conta os metodos push pop replacement.

Dei uma boa lida no código do GoRouter e vou tentar ajudar nisso nesta semana, pois ao meu ver, o context deve ser usado em views, mas podemos e deveriamos navegar em um controller por exemplo, caso contrario teriamos um return pra page novamente só pra usar uma navegação com context.

Vou tentar contribuir novamente, bom saber que está de volta e empenhado.
Boa sorte

**Comment by guillaumefielddata** [comment]
We are in the process of evaluating state managers for flutter for our new application. Are there any updates as of when Getx5 will release? I saw an update from @jonataslaw nearly 2 months ago that GetX5 was supposed to release within hours..

**Comment by yang-lile** [comment]
> We are in the process of evaluating state managers for flutter for our new application. Are there any updates as of when Getx5 will release? I saw an update from @jonataslaw nearly 2 months ago that GetX5 was supposed to release within hours..

Using riverpod would be a great option, it is one of flutter favorite packages.

**Comment by Linkadi98** [comment]
> We are in the process of evaluating state managers for flutter for our new application. Are there any updates as of when Getx5 will release? I saw an update from @jonataslaw nearly 2 months ago that GetX5 was supposed to release within hours..

Consider to using riverpod, it has been there for a while and it's pretty good. When GetX doesn't receive maintainance I switch to riverpod and stick to this package for 2 big projects.

**Comment by DevTello** [comment]
try this: [https://pub.dev/packages/stacked]. The author has built a lot of production apps with it, has his own website, and at least keeps his package up to date.

**Comment by wangjinbiao1985** [comment]
The following versions are relatively stable, with fewer bugs in the lower versions.
Getx 4.6.5 ~ Flutter 3.10
Getx 4.6.6 ~ Flutter 3.13

It is recommended not to upgrade Flutter 3.16 and higher.

Finally, Getx is not recommended for new projects.

**Comment by jakobleck** [comment]
> The following versions are relatively stable, with fewer bugs in the lower versions. Getx 4.6.5 ~ Flutter 3.10 Getx 4.6.6 ~ Flutter 3.13
> 
> It is recommended not to upgrade Flutter 3.16 and higher.

Any references or concrete reasons for not updating to Flutter 3.16? That would be quite a shocker...

**Comment by GotJimmy** [comment]
> we're on Flutter 3.16 , developing a relatively big enterprise app we didn't see any major issues linked to GetX at all.

Same here. Works great!

**Comment by ruanmax** [comment]
Whats is the future of Package? Will be release new version or will be descontinued ?

**Comment by DevTello** [comment]
It only has a future if forked and maintained by those who care. This package deserves a brighter future.

**Comment by ehangwork** [comment]
Looking at it every day, I still haven't waited for the release of the GetX 5. 
Can the author provide specific progress or follow-up arrangements?

**Comment by lilong-git** [comment]
when will the new version be published?

**Comment by MahdiForughi** [comment]
looking forward to new version of your awesome work.

**Comment by kellvembarbosa** [comment]
> looking forward to new version of your awesome work.

I really don't know if @jonataslaw will send the update, I really like the package, even more so because it was made by a Brazilian like me, but it disappeared in the last few months...

This makes me wonder a lot if he is overloaded with the package

**Comment by tariqarbi03** [comment]
@jonataslaw please tell us about getx 5 version??? Everyone is waiting of your response...

**Comment by jgirardet** [comment]
It's free opensource. you always know when you decide to use it what could  happen next.
The author doesn't owe us anything at all. Please stop spamming him.
He's for sur well aware of it, and if he doesn't respond we must respect him and let him quiet.
Everyone is welcomed to  fork like this guy did : https://github.com/Aniketkhote/refreshed/

**Comment by Aniketkhote** [comment]
> It's free opensource. you always know when you decide to use it what could happen next. The author doesn't owe us anything at all. Please stop spamming him. He's for sur well aware of it, and if he doesn't respond we must respect him and let him quiet. Everyone is welcomed to fork like this guy did : https://github.com/Aniketkhote/refreshed/

Hey there,

I appreciate you highlighting my contributions to the GetX project and bringing attention to the work I've been involved in. Your encouragement and acknowledgment are truly valued.

While I may not have the expertise of the original author of GetX, I'm passionate about contributing to the community and ensuring that the project continues to thrive. Your encouragement motivates me to keep pushing forward despite the challenges I face.

I completely agree that collaboration is key to making GetX better and more performant for everyone. I'm grateful for any help and guidance from the community, and I'm open to working together to address issues and implement improvements.

Let's continue to support each other and work towards making GetX the best it can be. Together, I'm confident we can overcome any obstacles and create something truly remarkable.

Thank you again for your support and encouragement.

**Comment by giantss** [comment]
looking forward to new version of your awesome work.

**Comment by laterdayi** [comment]
![image](https://github.com/jonataslaw/getx/assets/122137647/1e84ddcf-d67c-4737-9a85-30e6617eb4ee)
https://pub.dev/packages/refreshed

**Comment by fisforfaheem** [comment]
Dear Please ask community to help you out and release the next beta verison

**Comment by asterd** [comment]
Use https://pub.dev/packages/refreshed
Is getx5 based, the migration Is pretty simple and the progress are awesome.

**Comment by hayatbangash55** [comment]
Awesome package! I've been using it for three years, but it seems like the support for this package hasn't been very good for the last five to six months. Additionally, in the developer community, many developers spread rumors as if it's going to end badly. @jonataslaw should interact with the users of this package as soon as possible. If not, we can migrate to another state management system in a timely manner.

**Comment by fisforfaheem** [comment]
refreshed is good, but i want official GETX to update atleast just update some dependencies so we feel secure and stable.

**Comment by fisforfaheem** [comment]
> ![image](https://private-user-images.githubusercontent.com/122137647/313901138-1e84ddcf-d67c-4737-9a85-30e6617eb4ee.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUwNzA5ODUsIm5iZiI6MTcxNTA3MDY4NSwicGF0aCI6Ii8xMjIxMzc2NDcvMzEzOTAxMTM4LTFlODRkZGNmLWQ2N2MtNDczNy05YTg1LTMwZTY2MTdlYjRlZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwN1QwODMxMjVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NDNjODY0MmRlOWQ1YzcyODQyMDg2MTY1NzNkYTg5NTM3NmIxNTY5Y2VlZGZiZWEwODk1NTYxYTU1MTcyMzc1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8KkrD55Kvj5zI1fRhUILECUQbo9f9GvBQOHNPzG03zo) https://pub.dev/packages/refreshed

kindly contribute to GETX as well

**Comment by fisforfaheem** [comment]
Dear Dev, I know i am asking too much, but my whole project is stuck for months due to GETX no updating... kindly upgrade some dependencies atleast and give a minor update

**Comment by fisforfaheem** [comment]
We don't want to use refreshed, thr release candidate is working fine, just
needs some Minor updates..

**Comment by fisforfaheem** [comment]
[5.0.0-release-candidate-6] can it be released given the pull requests are tested and working for flutter 3.22

**Comment by nugrahazzz** [comment]
In the meantime, it's better to use the previous version of Flutter than the latest version that brings problems

**Comment by fisforfaheem** [comment]
> In the meantime, it's better to use the previous version of Flutter than the latest version that brings problems

No, u can use latest: get:
    git:
      url: https://github.com/Aniketkhote/getx.git
      ref: master

**Comment by mohsinnaqvi606** [comment]
Is it time to finally stop using GetX since it seems like GetX is not properly maintained anymore and GetX 5 is never going to be released? To be on the safe side, it seems like a fair step. @jonataslaw You have committed the last two times; @Aniketkhote what is your opinion on this?

**Comment by nugrahazzz** [comment]
@fisforfaheem how about get_storage? get_storage dependency on getx

**Comment by fisforfaheem** [comment]
@mohsinnaqvi606 brother the dev seems to be active, just not reluctant to release updates..


However our best bet is to depend on the solution i provided and if everyone uses it, and then find issues it would help a lot for further development.

**Comment by mohsinnaqvi606** [comment]
@fisforfaheem what is your's solution?

**Comment by fisforfaheem** [comment]
> > In the meantime, it's better to use the previous version of Flutter than the latest version that brings problems
> 
> No, u can use latest: get:
>     git:
>       url: https://github.com/Aniketkhote/getx.git
>       ref: master

Us this instead of fixed getX in pubspec.yaml

**Comment by giantss** [comment]
Don't pay too much attention to some people's gossip, concentrate on doing what you like and think is right. Come on, brother, I support you.
@jonataslaw

**Comment by tariqarbi03** [comment]
@jonataslaw any update on package?????

**Comment by alamakbar-sm** [comment]
> Don't pay too much attention to some people's gossip, concentrate on doing what you like and think is right. Come on, brother, I support you. @jonataslaw

what gossip? 👀

**Comment by wxqqh** [comment]
Getx is one of the most niubibility Flutter tool libraries, and hope Getx can stay true to its original intentions, continuously improve, and do what is right and beneficial to the Flutter ecosystem

**Comment by WMOH-DEV** [comment]
I'm very, very disappointed with this package. It's the simplest and easiest one, with great helpers, tools, and everything you need. But unfortunately, its future is very ambiguous, and you can't rely on it. No updates, no contributors, and no clear versioning

**Comment by fisforfaheem** [comment]
@jonataslaw kindly clear his doubts

**Comment by giantss** [comment]
I believe everyone shouldn't be so quick to jump to conclusions. The author just needs more time to complete his plan. In fact, the author hasn't abandoned GetX; he is working hard on another direction, but some people just haven't noticed. Encouragement from the community is what motivates the author to keep going, whereas criticism will only have a negative effect.

<img width="907" alt="image" src="https://github.com/user-attachments/assets/20297f39-9849-40f0-bedb-7657e2ffd592">

<img width="1320" alt="image" src="https://github.com/user-attachments/assets/1871990e-c0d7-4e14-a909-7f292244893a">

**Comment by shehabhy1** [comment]
i appreciate your clarification @giantss  so please don't have a look on the other because there is here a wave of criticism and clueless of the package and there is a lot of developers here have raised some claims and demands to not use GetX as it is not a reliable package and use another state management our developers here and to be precise most of them think that getX is a statemanagement only😂 so for that no worries i trust in the creator of the package and very soon will be considered as flutter favorite🤗

**Comment by ibhanu** [comment]
GetX will remain the most liked package on pub.dev
You have made dev time so less for MVP building.

IMHO: Let him cook

**Comment by fisforfaheem** [comment]
> I believe everyone shouldn't be so quick to jump to conclusions. The author just needs more time to complete his plan. In fact, the author hasn't abandoned GetX; he is working hard on another direction, but some people just haven' t noticed. Encouragement from the community is what motivates the author to keep going, whereas criticism will only have a negative effect.
> 
> <img alt="image" width="907" src="https://private-user-images.githubusercontent.com/13778798/369247610-20297f39-9849-40f0-bedb-7657e2ffd592.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4MzU4NDIsIm5iZiI6MTczMDgzNTU0MiwicGF0aCI6Ii8xMzc3ODc5OC8zNjkyNDc2MTAtMjAyOTdmMzktOTg0OS00MGYwLWJlZGItNzY1N2UyZmZkNTkyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA1VDE5MzkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2ZmM0NjE4MzZkOGZlOGIwN2ZmYzMwNWI1NzA4OWRiYjhmNGI5NGYyNjc2Nzg4OTc2MTVhOTg1MDc2ODVlYzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4XeWu8RtF_CQCx6nlLem08oQGD6skdTPEyxGvP48eAo"> <img alt="image" width="1320" src="https://private-user-images.githubusercontent.com/13778798/369247571-1871990e-c0d7-4e14-a909-7f292244893a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4MzU4NDIsIm5iZiI6MTczMDgzNTU0MiwicGF0aCI6Ii8xMzc3ODc5OC8zNjkyNDc1NzEtMTg3MTk5MGUtYzBkNy00ZTE0LWE5MDktN2YyOTIyNDQ4OTNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA1VDE5MzkwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRiNWZhNDNjNjFmZWZhNDE1MTQzMWNlN2UyNDZhYTlkYTlkOTEyY2ExZjc3MTlhNzkxOGRlNmI3NjhjMTUxM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ibp6wS6hqVOj6072s9QrWYy1ibBVcGJhVNLMxyVmq4E">

what about last 1 month

**Comment by raghavendra-k-j** [comment]
Can't you focus primary on state management and build it as a separate package, we have a better solution than getx for all other features you provide. rx widgets is the only need so we are free to use any package for any purpose

**Comment by supermnemonic** [comment]
Yeah i agree. Getx better focus state management. Thats it. Plus their rx is so simple and thats good.

Storage? Network? Localization? I dont need it. I like native or other libs for that.

**Comment by fisforfaheem** [comment]
Yes agree with u we just need next getX no other useless stuff

**Comment by agil3st** [comment]
This feels like this package is abandoned by the author. since we don't receive any updates or even commits in past months. @jonataslaw please at least communicate with us OR share the timeline of this stable version release. Hoping that we can use it in the next month if possible.

**Comment by alex-lechner** [comment]
Would be great to get some progress updates if possible @jonataslaw

**Comment by axu6705** [comment]
Any news?

**Comment by colbymaloy** [comment]
yeah in my experience getx is easy to understand for state management and dependency injection, so i use it for that. but most of the other stuff i dont really see a need to use. especially routing - as it feels like i understand flutter's own navigation LESS after using Get's navigation.

**Comment by NebiSarigul** [comment]
Any news?

**Comment by fisforfaheem** [comment]
@jonataslaw/getx ***@***.***> please 🥺 reply back

**Comment by fisforfaheem** [comment]
😔

**Comment by shalabycr7** [comment]
is this still maintained?

**Comment by shalabycr7** [comment]
> @shalabycr7 nobody knows but from the (non) activity you can infer it's not maintained anymore. I'm currently using it for an enterprise application and there are a lot of known - and still unresolved - bugs like the proper initialization and disposal of controllers. The maintainer received a generous funding offer from the community to keep it alive which he kind of declined and promised that he `will never abandon` it and to `organize himself` for this package (see #3159 ). It's a great package - but in my case I have to move on. 
> 
> For anyone who's interested in a replacement, maybe [bloc](https://github.com/felangel/bloc) might be a good fit (but personally I do not have any experience yet and will evaluate it within the next weeks). 
> 

thank you for the updated info, I also see another project named "refreshed" I think that is trying to keep up getx update I think I will give it a try

**Comment by fisforfaheem** [comment]
Till then you can use the latest version : https://github.com/Aniketkhote/getx/tree/master

**Comment by fisforfaheem** [comment]
The developer should communicate atleast @jonas

**Comment by whevether** [comment]
> The developer should communicate atleast [@jonas](https://github.com/jonas)

You are really poisonous， Since you are so persistent, why not try reading the source code or writing something similar on your own. I can still grow on my own

**Comment by nugrahazzz** [comment]
I think it's time to update the stable version because it's too outdated. Flutter has released its latest version (3.32.0) maybe there will be more bugs in the old version of getx (stable version)

**Comment by wanwenfeng4798** [comment]
> I think it's time to update the stable version because it's too outdated. Flutter has released its latest version (3.32.0) maybe there will be more bugs in the old version of getx (stable version)

There are too few contributors. And the author doesn't seem to pay much attention to it. Suggest converting new or large-scale projects to Bloc and go_router, as these two are more actively maintained

**Comment by wanwenfeng4798** [comment]
It's really exaggerated. The first beta will be released in 2023, and it's already mid-2025 this year. There is neither an official version of the planned route nor active maintenance. Yes, the author is not full-time. I haven't paid for it either. But I haven't paid for the same "bloc" and "go_router" either. But they are very proactive in maintaining it.

**Comment by maxfrees** [comment]
https://pub.dev/packages/refreshed    It has also been discontinued. I don't know why.  @Aniketkhote

**Comment by Aniketkhote** [comment]
@maxfrees 

Thanks for bringing this up. I actually discontinued the package around two months ago. Since then, only two people have noticed, which gives me a rough idea of how much interest there really was.

I was actively trying to improve and maintain the package, but unfortunately, there wasn’t much support or engagement from the community. Projects like these take time and effort, and it's hard to sustain them without feedback, contributions, or even just interest. It’s not something one person can keep up with alone.

That said, I'm still happy to see it was useful to some of you. If anyone wants to take it forward, I’d be open to discussing that.

**Comment by maxfrees** [comment]
Actually everyone is paying attention to it, but getx hasn't been updated for a long time, and refreshed has only been updated for two months, so everyone thought it was still being actively maintained.

**Comment by maxfrees** [comment]
Anyway, thank you for your efforts, and I hope the flutter community will get better and better.

**Comment by fisforfaheem** [comment]
@Aniketkhote 
Wait… when did you tell us that you discontinued this?
We were still hoping. Still waiting.
Every time Flutter pushed a new update, we looked forward to yours too — your updates, your support, your commitment. You always delivered. You made the impossible easy for so many of us. For developers like me who don’t have deep expertise, your work was a lifeline.

You said you’d still try — even just a few hours a week. That hope kept us going.
Please don’t let it end like this.
Not without giving us a chance to help.
This project matters so much more than you may realize.

Please — if time is short, maybe open a small funding option? Even a donation link.
I promise I’ll be the first to contribute.

Just don’t cancel this.
Not yet.
Not like this.

From one of many devs whose work literally depends on what you built. 🙏💔

**Comment by chjsun** [comment]
什么时候发布新版本呢？

**Comment by whevether** [comment]
> 什么时候发布新版本呢？

It's unfinished. brother. 5.0 has not yet entered pre release from 24 years ago. Carefully consider the new project

**Comment by alex-lechner** [comment]
@fisforfaheem just switch to Bloc man. It's way more reliable / resilient, has better documentation and is also reactive: https://bloclibrary.dev/

**Comment by whevether** [comment]
> [@jonataslaw](https://github.com/jonataslaw) Kindly Please give an update, It's been months of silence please do a beta release

This library is already abandoned. I learned Bloc in three days. I suggest moving to Bloc or Riverpod, i use template https://github.com/whevether/flutter_bloc_template

**Comment by mahanpyk** [comment]
While reading recent comments, I noticed that some individuals, instead of contributing constructively to improve the project, are engaging in destructive criticism and mockery, which is truly disappointing. Unfortunately, we’re dealing with unreasonable and biased developers who lack originality and problem-solving skills yet make grandiose claims. They expect others to provide fully customized solutions for their projects without contributing themselves. I want to be part of the solution by helping address this issue, supporting a package I proudly use and consider the best in its field. Additionally, I aim to silence some of these unproductive critics through meaningful contributions. Lastly, I’d appreciate it if you could share, even briefly, any remaining issues or tasks you haven’t had time to complete. I’m eager to assist and help bring them to completion. Thank you!
@jonataslaw

**Comment by nugrahazzz** [comment]
It would be highly beneficial if get_packer, get_storage, and GetX 5 could function properly in a unified project. Currently, the outdated version of get_storage acts as a bottleneck. I hope to see an update that enables full compatibility across these three core packages @jonataslaw

**Comment by jonataslaw** [comment]
I think so, I get 14.000.000% speed I get_packer over json, would be blazing fast as storage and route restoration feature

**Comment by fisforfaheem** [comment]
any updates on the project brother for 2026

**Comment by maxmoridev** [comment]
@jonataslaw any update on making GetX5 compatible with Flutter v3.44.0?
 .. are you planning to make it compatible?

**UPDATE:**
if anyone have problem, 
 the temporary solution is to add _"import 'package:flutter/cupertino.dart';"_
in
lib/get_navigation/src/root/get_root.dart

------>

import 'package:flutter/cupertino.dart'; // <-- import Cupertino for the new Flutter version
...
import 'package:flutter/material.dart';   
...
switch (matchingTransition) {
  case CupertinoPageTransitionsBuilder():   // <-- row 450, not resolved if missing Cupertino import
    return Transition.cupertino;
  ...
}

**Comment by gauravmehta13** [comment]
@maxmoridev 
use this version


```yaml
dependencies:
  getxtra:
```

Import get in files that it will be used:

```dart
import 'package:getxtra/get.dart';
```

**Comment by yanashenyang** [comment]
Always resolutely resist this library

**Comment by Aniketkhote** [comment]
@yanashenyang use getxify

**Comment by fisforfaheem** [comment]
Could you please also maintain a branch of GetX to match getxify as well?
This will ensure we have both options available for the community.

**Comment by nugrahazzz** [comment]
In my opinion, Zenify is the best alternative because all the issues in GetX—especially the critical ones—have been fixed (as you can see in the changelog).

I guarantee you’ll understand why I recommend this once you’ve tried it on medium- to large-scale projects.

https://pub.dev/packages/zenify

**Comment by Aniketkhote** [comment]
@nugrahazzz you can check this also it has solved nearly 100+ issues and added nearly 450+ tests

https://github.com/Aniketkhote/getxify

**Comment by fisforfaheem** [comment]
Hi Everyone,

I think the main issue is naming. We need GETX2 so it feels natural, just
do AB testing!

Best regards,
Faheem Ahmad
